tests/unit/test_util.py

```diff
@@ -0,0 +1,86 @@
+from unittest.mock import Mock, patch
+
+import httpx
+import pytest
+
+from entitysdk.common import ProjectContext
+from entitysdk.util import make_db_api_request
+
+
+@pytest.fixture
+def mock_client():
+    return Mock()
+
+
+def test_make_db_api_request(mock_client: Mock):
+    url = "http://localhost:8000/api/v1/entity/person"
+
+    make_db_api_request(
+        url=url,
+        method="POST",
+        json={"name": "John Doe"},
+        parameters={"foo": "bar"},
+        token="123",
+        project_context=ProjectContext(
+            project_id="123",
+            virtual_lab_id="456",
+        ),
+        http_client=mock_client,
+    )
+
+    mock_client.request.assert_called_once_with(
+        method="POST",
+        url=url,
+        headers={"project-id": "123", "virtual-lab-id": "456", "Authorization": "Bearer 123"},
+        json={"name": "John Doe"},
+        params={"foo": "bar"},
+    )
+
+
+def test_make_db_api_request_with_none_http_client__raises(mock_client: Mock):
+    url = "http://localhost:8000/api/v1/entity/person"
+
+    mock_response = Mock()
+    mock_response.raise_for_status.side_effect = httpx.HTTPStatusError(
+        message="404 Not Found", request=Mock(), response=Mock(status_code=404)
+    )
+    mock_client.request.return_value = mock_response
+
+    with pytest.raises(httpx.HTTPError, match="person"):
+        make_db_api_request(
+            url=url,
+            method="POST",
+            json={"name": "John Doe"},
+            parameters={"foo": "bar"},
+            project_context=ProjectContext(
+                project_id="123",
+                virtual_lab_id="456",
+            ),
+            token="123",
+            http_client=mock_client,
+        )
+
+
+@patch("entitysdk.util.httpx")
+def test_make_db_api_request_with_none_http_client__client_none(mock_httpx: Mock):
```

Comment:
We could use `httpx_mock` from the `pytest-httpx` package if we need more features, but this seems enough